modules/imgcodecs/src/grfmt_tiff.cpp

```diff
@@ -710,32 +710,40 @@ bool  TiffDecoder::readData( Mat& img )
                                 bstart += (tile_height0 - tile_height) * tile_width0 * 4;
                             }
 
-                            for (int i = 0; i < tile_height; i++)
-                            {
-                                if (color)
-                                {
-                                    if (wanted_channels == 4)
-                                    {
-                                        icvCvt_BGRA2RGBA_8u_C4R(bstart + i*tile_width0*4, 0,
-                                                img.ptr(img_y + tile_height - i - 1, x), 0,
-                                                Size(tile_width, 1) );
-                                    }
-                                    else
+                            const int minimalBytesPerPacket = 1<<12;
+                            const int rowsPerPacket = std::max(1, static_cast<int>(minimalBytesPerPacket/(ncn * tile_width0 * sizeof(uchar))));
+                            const int rowsPacketsCount = divUp(tile_height, rowsPerPacket);
+                            const Range all(0, tile_height);
+                            parallel_for_(all, [=,&img](const cv::Range& range) -> void {
```

Comment:
> `[=,`

Capturing variables by value (through coping) is slow and there is no reason to do that (we don't need to expand lifetime).
Use lambda variables capturing by reference.

Reply:
That was one of my numerous attempts to avoid the perf regression. In that particular case, [&] or [=,&img] give roughly the same performance, because the=captures are just a few bytes. I can revert to [&], as code cleanup, though.